pkg/controllers/disruption/consolidation_test.go

```diff
@@ -1223,10 +1223,11 @@ var _ = Describe("Consolidation", func() {
 			// Forcefully assign lowest possible instancePrice to make sure we have atleast one instance
 			// that is lower than the current node.
 			cloudProvider.InstanceTypes[0].Offerings[0].Price = 0.001
-			cloudProvider.InstanceTypes[0].Offerings[0].CapacityType = v1beta1.CapacityTypeSpot
+			cloudProvider.InstanceTypes[0].Offerings[0].Requirements[v1beta1.CapacityTypeLabelKey] = scheduling.NewRequirement(
+				v1beta1.CapacityTypeLabelKey, v1.NodeSelectorOpIn, v1beta1.CapacityTypeSpot)
```

Comment:
Wondering how this doesn't error if we ensure that zone and capacity type must be set in these requirements.

Reply:
This is just overwriting the existing capacity type, it is set by default by the fake cloudprovider